assemblyline_core/updater/helper.py

```diff
@@ -121,7 +121,8 @@ def process_image(image):
         auth = f"Basic {b64encode(upass.encode()).decode()}"
 
     server, image_name = process_image(searchable_image)
-    registry = REGISTRY_TYPE_MAPPING[service_config.docker_config.registry_type]
+    registry_type = service_config.docker_config.registry_type or system_config.services.preferred_registry_type
```

Comment:
This is unnecessary, there should always be a value in service_config.docker_config.registry_type since the add/register operation was hijacked to always provide the prefered value.